usr/lib/cca_stdll/cca_specific.c

```diff
@@ -3310,6 +3323,117 @@ CK_RV token_specific_set_attrs_for_new_object(STDLL_TokData_t *tokdata,
 }
 #endif /* NO_PKEY */
 
+typedef struct {
+    CK_BBOOL card_level_set;
+    struct {
+        unsigned int ver;
+        unsigned int rel;
+        unsigned int mod;
+    } min_card_version;
+} cca_min_card_version_t;
+
+#define CCA_STATCCA_CCA_VERSION_OFFSET         24
+#define CCA_STATCCA_CCA_VERSION_LENGTH          8
+#define MAX_INT                                0xFFFFFFFF
+
+/*
+ * Called from within cca_iterate_adapters() handler function, thus no need to
+ * obtain  CCA adapter lock
+ */
+static CK_RV cca_get_adapter_version(cca_min_card_version_t *data)
+{
+    unsigned char rule_array[CCA_RULE_ARRAY_SIZE] = { 0, };
+    long return_code, reason_code, rule_array_count, verb_data_length;
+    unsigned int ver = MAX_INT, rel = MAX_INT, mod = MAX_INT;
+    char ccaversion[8];
+
+    memcpy(rule_array, "STATCCA ", CCA_KEYWORD_SIZE);
+    rule_array_count = 1;
+    verb_data_length = 0;
+    dll_CSUACFQ(&return_code, &reason_code,
+                NULL, NULL,
+                &rule_array_count, rule_array,
+                &verb_data_length, NULL);
+
+    if (return_code != CCA_SUCCESS) {
+        TRACE_ERROR("CSUACFQ (STATCCA) failed. return:%ld, reason:%ld\n",
+                    return_code, reason_code);
+        return CKR_FUNCTION_FAILED;
+    }
+
+    memcpy(ccaversion, &rule_array[CCA_STATCCA_CCA_VERSION_OFFSET],
```

Comment:
This leaves ccaversion non-zero-terminated ! 
Make sure to add `ccaversion[CCA_STATCCA_CCA_VERSION_LENGTH] = '\0';`

Reply:
Done.